init

confa-hub opened the pull request:
init project

confa-hub committed:
init

init project

EkimEvgeny commented on .idea/.gitignore:
```diff
@@ -0,0 +1,5 @@
+# Default ignored files
```
Папку .idea не нужно коммитить

confa-hub replied:
Исправлено

EkimEvgeny commented on config-application.json:
```diff
@@ -0,0 +1,28 @@
+    "port": "5432",
+    "pathPgDump": "C:/Program Files/PostgreSQL/14/bin/pg_dump.exe",
+    "pathSaveDatabase": "E:/node.js/Dev/creator-backup-copies/src/tmp/DataBaseBackup",
+    "databaseSaveToTmp": "E:/node.js/Dev/creator-backup-copies/src/tmp/BackupDataBase.backup"
```
Зачем два почти одинаковых параметра нужны "directoryTMPPath" и "databaseSaveToTmp"? Если причин существенных нет, то нужно сделать один параметр "tempDirectoryPath" и указывать путь к папке, где будут храниться временные файлы.

confa-hub replied:
Исправлено

EkimEvgeny commented on package.json:
```diff
@@ -0,0 +1,83 @@
+    "dotenv": "^16.0.3",
+    "file-saver": "^2.0.5",
+    "jszip": "^3.10.1",
+    "nodejs-zip": "^0.1.0",
```
Лишняя библиотека. Нужно удалить

confa-hub replied:
Исправлено

EkimEvgeny commented on package.json:
```diff
@@ -0,0 +1,83 @@
+    "axios": "^1.1.3",
+    "dayjs": "^1.11.6",
+    "dotenv": "^16.0.3",
+    "file-saver": "^2.0.5",
```
Используется ли эта библиотека? Если нет, то нужно удалить. Если да, то нужно объяснить зачем?

confa-hub replied:
Исправлено

EkimEvgeny commented on package.json:
```diff
@@ -0,0 +1,83 @@
+  },
+  "dependencies": {
+    "@getvim/execute": "^1.0.0",
+    "@nestjs/axios": "^0.1.0",
```
Нужно удалить

confa-hub replied:
Исправлено

EkimEvgeny commented on package.json:
```diff
@@ -0,0 +1,83 @@
+    "@nestjs/core": "^9.0.0",
+    "@nestjs/platform-express": "^9.0.0",
+    "async-lock": "^1.4.0",
+    "axios": "^1.1.3",
```
Нужно удалить

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+export class AppService {
+
+  differenceTIme: number = 1;
+  private obj: any;
```
Что это за переменная? Почему тип any? (должен быть конкретный тип данных). К переменным-полям классов обязательно нужны комментарии.

confa-hub replied:
Данная переменная переименована в configData. Тип данной переменной any потому что она инициализируется методом Json.parse который тоже возращает any.

confa-hub replied:
Переменная удалена

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+@Injectable()
+export class AppService {
+
+  differenceTIme: number = 1;
```
Написано неправильно - должно быть differenceTime. Также по смыслу не совсем правильно и не дает понимания, зачем эта переменная нужна. Логичнее было  назвать ее как-то вроде "firstBackupDelay". Непонятно, почему это число, и почему оно устанавливается в 1 - нужно расписать в комментарии.

EkimEvgeny replied:
Также непонятно, зачем эта переменная объявлена как поле класса, а не инициализируется и используется только в методе инициализации. Нужно ее оставить только в методе класса.

confa-hub replied:
Данная переменная переименована firstBackupDelay. Объявлена  в начале класса для использования в других методах и имеет default значение 1. Так же эта переменная инициализируется методом getNumberOfTimes.

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+   */
+  async init() {
+
+    const data = fs.readFileSync("E:/node.js/Dev/creator-backup-copies/config-application.json", "utf8");
```
Здесь должен стоять относительный путь. Так будет работать только на твоем компьютере.  В программе, вообще, нигде нельзя использовать полные пути. Полные пути только в конфигах.

EkimEvgeny replied:
Название переменной должно отражать, что эта переменная хранит. Здесь нормально подойдет "configData" или подобное

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+    const data = fs.readFileSync("E:/node.js/Dev/creator-backup-copies/config-application.json", "utf8");
+    this.obj = JSON.parse(data);
+
+
```
Лишний отступ

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+  async init() {
+
+    const data = fs.readFileSync("E:/node.js/Dev/creator-backup-copies/config-application.json", "utf8");
+    this.obj = JSON.parse(data);
```
Название переменное должно отражать, что это такое

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+    const pathsFileForArchive = this.obj["pathFileOrFolderForArchive"];
+    const logFile = this.obj["logFile"];
+
+
```
Лишний отступ

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+
+
+    const pathsFileForArchive = this.obj["pathFileOrFolderForArchive"];
+    const logFile = this.obj["logFile"];
```
Название должно быть понятнее. Здесь понятнее будет logFilePath

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+    }
+
+    this.fileService.createFileLog(logFile);
+    this.differenceTIme = this.fileService.lastDateBackupDifference(logFile, this.numberOfTimes());
```
Где используется эта переменная? Вроде бы нигде. Работает ли сейчас создание первого бэкапа после запуска программы?

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+      setInterval(async () => {
+        await this.fileOperations(pathsFileForArchive, zip);
+      }, this.numberOfTimes());
+    }, 2);
```
Почему стоит интервал 2?

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+
+  /**
+   * Метод для архивации,отправки на ЯндексДиск и удаления файлов.
+   * Создан чтобы убрать дублирования кода
```
Писать комментарии, что метод нужен для того, чтобы убрать дублирование кода не нужно. Все методы нужны только для того, чтобы код не дублировать.

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+   * @param zip
+   * @private
+   */
+  private async fileOperations(pathsFileForArchive, zip) {
```
Из названия метода непонятно, что он делает. Нормально было бы что-то вроде "createFullBackup" или наподобии.
pathsFileForArchive - по-английски очень криво написано. Нужно filePathsForArchive или archiveFilePaths

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+   * Получить частоту получения бэкапов
+   */
+  numberOfTimes(): number {
+    let times = this.obj["backup_frequency"];
```
Название параметра backup_frequency должно быть без подчеркивания

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+      times = 1;
+    }
+
+    const number = 86400000 / this.obj["backup_frequency"];
```
Константу 864000000 лучше разбить на несколько, например 60 * 60 * 24 * 1000 - так намного понятнее, что это за число. У переменной должно быть осмысленное название.

confa-hub replied:
Исправлено

EkimEvgeny commented on src/app.service.ts:
```diff
@@ -0,0 +1,94 @@
+  /**
+   * Получить частоту получения бэкапов
+   */
+  numberOfTimes(): number {
```
У функции должно быть осмысленное название, которое соответствует тому, что эта функция делает

confa-hub replied:
Исправлено

EkimEvgeny commented on src/file-manager/dto/fileBackup.dto.ts:
```diff
@@ -0,0 +1,12 @@
+/**
+ * DTO для хранения имени файла и его пути
+ */
+export class FileBackup{
```
Нет комментариев у полей класса.

confa-hub replied:
Исправлено

EkimEvgeny commented on src/file-manager/file-manager.module.ts:
```diff
@@ -0,0 +1,8 @@
+  providers: [FileManagerService],
+  exports: [FileManagerService]
+})
+export class FileManagerModule {}
```
Нет комментария у класса.

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+import { Injectable } from "@nestjs/common";
+import * as fs from "fs";
+
```
Убрать пустые строки

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+   */
+  async isFolderYandexDisk(): Promise<boolean> {
+    try {
+      let data = fs.readFileSync('E:/node.js/Dev/creator-backup-copies/config-application.json', 'utf8');
```
Нельзя использовать абсолютные пути!!!

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+  /**
+   * Метод проверяет существует ли папка на ЯндексДиске
+   */
+  async isFolderYandexDisk(): Promise<boolean> {
```
Из коммента непонятно существование какой папки проверяется. Соответственно и название метода непонятное.

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+      let data = fs.readFileSync('E:/node.js/Dev/creator-backup-copies/config-application.json', 'utf8');
+      const configApplication = JSON.parse(data);
+      let tokenYandexDisk = configApplication['tokenYandexDisk']
+      const directoryTMPPath = configApplication[`directoryTMPPath`]
```
Зачем здесь эта константа, если она не используется?

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+      const directoryTMPPath = configApplication[`directoryTMPPath`]
+      await meta.get(tokenYandexDisk, this.folder);
+    } catch (error) {
+      return false;
```
Если выдалось исключение, то его нужно, как минимум записать в консоль или лог.

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+   */
+  private async createFolderYandexDisk(): Promise<boolean> {
+    try {
+      let data = fs.readFileSync('E:/node.js/Dev/creator-backup-copies/config-application.json', 'utf8');
```
Нельзя использовать абсолютные пути!!!

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+  private async createFolderYandexDisk(): Promise<boolean> {
+    try {
+      let data = fs.readFileSync('E:/node.js/Dev/creator-backup-copies/config-application.json', 'utf8');
+      const configApplication = JSON.parse(data);
```
Должен быть модуль и сервис конфигурации, в котором парсинг файла будет просходить только при инициализации сервиса.

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+        await waitFor('done', emitter);
+
+    } catch (error) {
+      console.error("Ошибка ==== " + error);
```
нужно посмотреть, как делаются логи в nestJS и везде использовать логирование, как в nestJS. https://docs.nestjs.com/techniques/logger

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+   * Медор использует Множество действий (Создание папки и отправку файла)
+   */
+  async uploadYandexDisk() {
+    let data = fs.readFileSync('E:/node.js/Dev/creator-backup-copies/config-application.json', 'utf8');
```
Абсолютные пути в программе под запретом!!!

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+  }
+
+  /**
+   * Медор использует Множество действий (Создание папки и отправку файла)
```
Из коммента непонятно, что конкретно делает файл. Из названия непонятно, что именно загружает этот метод.

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+    let data = fs.readFileSync('E:/node.js/Dev/creator-backup-copies/config-application.json', 'utf8');
+    const configApplication = JSON.parse(data);
+    const logFile = configApplication['logFile']
+    this.folder = dayjs().format("YYYY-MM-DD HH:mm");
```
название папки не может включать в себя двоеточие. Формат даты не русский. Нормально будет иметь формат "DD.MM.YYYY-HH mm" или даже "HH mm-DD.MM.YYYY".

confa-hub replied:
Исправлено

EkimEvgeny commented on src/yandex-disk/services/yandex-disk.service.ts:
```diff
@@ -0,0 +1,158 @@
+    }
+
+    for (const backup of await this.getAllFilesTmpDir()) {
+      const timeFileBackup = dayjs().format("YYYY-MM-DD HH:mm");
```
название файла не может включать в себя двоеточие. Формат даты не русский. Нормально будет иметь формат "DD.MM.YYYY-HH mm

confa-hub replied:
Исправлено

EkimEvgeny commented on src/zip-input-stream/zip-input-stream.module.ts:
```diff
@@ -0,0 +1,9 @@
+  providers: [ZipInputStreamService],
+  exports: [ ZipInputStreamService]
+})
+export class ZipInputStreamModule {}
```
Нет комментария у класса

confa-hub replied:
Исправлено

EkimEvgeny commented on src/zip-input-stream/services/zip-input-stream.service.ts:
```diff
@@ -0,0 +1,102 @@
+  /**
+   * Создаёт бэкап базы данных postgres
+   */
+  async backupDataBase(){
```
Непонятно, какое отношение этот метод имеет к этому классу. Нужно перенести его в более уместное место.

confa-hub replied:
Исправлено

EkimEvgeny commented on config-application.json:
```diff
@@ -0,0 +1,28 @@
+  "logFile" : "./lastDateBackupLog.txt",
+  "tokenYandexDisk" : "y0_AgAAAAAidqsRAADLWwAAAADSQp5CFprCXZfzTFWiPX8xWFp5C6N-AcA",
+
+  "backup_frequency" : 23,
```
Название параметра должно быть без подчеркивания

confa-hub replied:
Исправлено

EkimEvgeny commented on package.json:
```diff
@@ -0,0 +1,83 @@
+    "@nestjs/common": "^9.0.0",
+    "@nestjs/core": "^9.0.0",
+    "@nestjs/platform-express": "^9.0.0",
+    "async-lock": "^1.4.0",
```
Вроде, библиотека не используется. Если так, то нужно удалить.

confa-hub replied:
Исправлено

EkimEvgeny commented on package.json:
```diff
@@ -0,0 +1,83 @@
+    "rxjs": "^7.2.0",
+    "wait-for-event": "^2.0.1",
+    "ya-disk": "^4.0.1",
+    "zip-dir": "^2.0.0"
```
Вроде библиотека не используется. Если так, то нужно удалить.

confa-hub replied:
Исправлено

confa-hub committed:
changes conversation 41

-delete spaces
-rename variables
-fix 2 ms setTimeout
-create module database-manager
-delete excess libraries
-add class comments
-add parameter in gitignore

confa-hub committed:
rename method

rename method getNumberOfTime on getTimeRepeatBackup

confa-hub committed:
rename method

getTimeRepeatBackup to getMillisecondsBetweenBackups

confa-hub committed:
add config-module and logger

add config-module and logger

confa-hub committed:
more micro fix

more micro fix

confa-hub committed:
last my push

last my push

confa-hub committed:
last my push v2

last my push v2
